src/main.cpp

```diff
@@ -2319,22 +2330,25 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
     // Construct the incremental merkle tree at the current
     // block position,
-    auto old_tree_root = view.GetBestAnchor();
+    auto old_sprout_tree_root = view.GetBestAnchor(SPROUT);
     // saving the top anchor in the block index as we go.
     if (!fJustCheck) {
-        pindex->hashAnchor = old_tree_root;
+        pindex->hashSproutAnchor = old_sprout_tree_root;
     }
-    ZCIncrementalMerkleTree tree;
+    ZCIncrementalMerkleTree sprout_tree;
     // This should never fail: we should always be able to get the root
     // that is on the tip of our chain
-    assert(view.GetAnchorAt(old_tree_root, tree));
+    assert(view.GetSproutAnchorAt(old_sprout_tree_root, sprout_tree));
 
     {
         // Consistency check: the root of the tree we're given should
         // match what we asked for.
-        assert(tree.root() == old_tree_root);
+        assert(sprout_tree.root() == old_sprout_tree_root);
     }
 
+    ZCSaplingIncrementalMerkleTree sapling_tree;
+    assert(view.GetSaplingAnchorAt(view.GetBestAnchor(SAPLING), sapling_tree));
+
```

Comment:
Should there be a check similar to 
```
        assert(sprout_tree.root() == old_sprout_tree_root);
```
but for sapling?

Reply:
Meh, it's not really necessary. We used to grab the old tree root for the undo structures but we're not doing that in Sapling.